### PR TITLE
Basic Kafka Streaming application setup.

### PR DESCRIPTION
**Pull request by jasebell**

### CODE REVIEW

File: deployment/kafkas3backup-stream-config.json.template [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+              "hostPort": 18082,
+              "protocol": "tcp"
+          }
+      ]
+    }
+  },
+  "mem": 2048,
+  "cpus": 1,
+  "instances": 1,
+  "id": "tp-streams@@NUMBER@@"
```
Comment (elisehuard) [inline deployment/kafkas3backup-stream-config.json.template]: do we go with identified numbers, of could we just have several instances?

File: project.clj.tmpm [hunk 2]
```diff
@@ -0,0 +1,21 @@
+(defproject kixi.kafka.s3.backup "0.1.0-SNAPSHOT"
+  :description "FIXME: write description"
+  :url "http://example.com/FIXME"
+  :license {:name "Eclipse Public License"
```
Comment (elisehuard) [inline project.clj.tmpm]: not sure about function of this file? should it be removed?

File: src/kixi/kafka/s3/core.clj [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+(defn broker-str [zkconf]
+  (let [zk-brokers (get-broker-list zkconf)
+        brokers (map (fn [broker] (str (get-in broker [:endpoints :plaintext :host]) ":" (get-in broker [:endpoints :plaintext :port])) ) zk-brokers)]
+    (if (= 1 (count brokers))
+      (first brokers)
+      (cstr/join "," brokers))))
+
+(defn process-data [data-in]
+  (let [incoming-message (shared/deserialize-message data-in)]
```
Comment (elisehuard) [inline src/kixi/kafka/s3/core.clj]: does this assume the incoming messages are gzipped?

File: project.clj.tmpm (hunk 2 above)
Comment (jasebell) [reply project.clj.tmpm]: Good catch, should have been removed.

File: src/kixi/kafka/s3/core.clj (hunk 3 above)
Comment (jasebell) [reply src/kixi/kafka/s3/core.clj]: correct, needs amending.

File: test/kixi/kafka/s3/backup_test.clj [hunk 4]
```diff
@@ -0,0 +1,7 @@
+(ns kixi.kafka.s3.backup-test
+  (:require [clojure.test :refer :all]
+            [kixi.kafka.s3.backup :refer :all]))
+
+(deftest a-test
```
Comment (elisehuard) [inline test/kixi/kafka/s3/backup_test.clj]: (I'll do a Bruce and suggest that this test be removed ;) )

**Review by elisehuard — COMMENTED** [review]
several small comments

File: deployment/kafkas3backup-stream-config.json.template (hunk 1 above)
Comment (jasebell) [reply deployment/kafkas3backup-stream-config.json.template]: Just went with how we've done things in the past. For ease I'd prefer to go with what we know.

File: deployment/kafkas3backup-stream-config.json.template [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{
+  "container": {
+    "type": "DOCKER",
+    "docker": {
+      "image": "mastodonc/tp-streams:git-55c906e3b35d",
+      "parameters": [
+         {"key": "log-driver", "value": "gelf"},
+         {"key": "log-opt", "value": "gelf-address=udp://logstash.vpc-vpc.kixi:12201"},
```
Comment (elisehuard) [inline deployment/kafkas3backup-stream-config.json.template]: we don't use gelf anymore, so the first 4 parameters should be removed

Comment (jasebell) [reply deployment/kafkas3backup-stream-config.json.template]: done

**Review by elisehuard — APPROVED** [review]

**Pull request merged**